test/lan_udev_auto_test.rb

```diff
@@ -19,61 +22,93 @@ module Yast
-      end
+  context 'when new style name is provided' do
+    it 'returns the new style name' do
+      expect( LanUdevAuto.getDeviceName( NEW_STYLE_NAME)).to be_equal NEW_STYLE_NAME
     end
+  end
 
-    context 'when old fashioned mac based name is provided' do
-      it 'returns corresponding new style name' do
-        expect( LanUdevAuto.getDeviceName( MAC_BASED_NAME)).to be_equal NEW_STYLE_NAME
-      end
-
-      it 'returns same result despite of letter case in mac' do
-        expect(
-          LanUdevAuto.getDeviceName( LCASE_MAC_NAME)
-        ).to be_equal LanUdevAuto.getDeviceName( MAC_BASED_NAME)
-      end
+  context 'when old fashioned mac based name is provided' do
+    it 'returns corresponding new style name' do
+      expect( LanUdevAuto.getDeviceName( MAC_BASED_NAME)).to be_equal NEW_STYLE_NAME
+    end
 
-      it 'returns given name if no known device is matched' do
-        expect( LanUdevAuto.getDeviceName( UNKNOWN_MAC_NAME)).to be_equal UNKNOWN_MAC_NAME
-      end
+    it 'returns same result despite of letter case in mac' do
+      expect(
+        LanUdevAuto.getDeviceName( LCASE_MAC_NAME)
+      ).to be_equal LanUdevAuto.getDeviceName( MAC_BASED_NAME)
     end
 
-    context 'when old fashioned bus id based name is provided' do
-      it 'returns corresponding new style name' do
-        expect( LanUdevAuto.getDeviceName( BUS_BASED_NAME)).to be_equal NEW_STYLE_NAME
-      end
+    it 'returns given name if no known device is matched' do
+      expect( LanUdevAuto.getDeviceName( UNKNOWN_MAC_NAME)).to be_equal UNKNOWN_MAC_NAME
+    end
+  end
 
-      it 'returns given name if no known device is matched' do
-        expect( LanUdevAuto.getDeviceName( UNKNOWN_MAC_NAME)).to be_equal UNKNOWN_MAC_NAME
-      end
+  context 'when old fashioned bus id based name is provided' do
+    it 'returns corresponding new style name' do
+      expect( LanUdevAuto.getDeviceName( BUS_BASED_NAME)).to be_equal NEW_STYLE_NAME
     end
 
-    context 'when provided invalid input' do
-      # TODO: should raise an exception in future
-      it 'returns given input' do
-        expect( LanUdevAuto.getDeviceName( INVALID_NAME)).to be_equal INVALID_NAME
-      end
+    it 'returns given name if no known device is matched' do
+      expect( LanUdevAuto.getDeviceName( UNKNOWN_MAC_NAME)).to be_equal UNKNOWN_MAC_NAME
     end
+  end
 
+  context 'when provided invalid input' do
+    # TODO: should raise an exception in future
+    it 'returns given input' do
+      expect( LanUdevAuto.getDeviceName( INVALID_NAME)).to be_equal INVALID_NAME
+    end
   end
+
 end
+
+describe "LanUdevAuto#Write" do
+
+  ATTR = "ATTR{address}"
+  VALUE = "aa:BB:cc:DD:ee:FF"
+  NAME = "custom-name"
+
+  it "writes MAC in lowercase" do
+    udev_rules = [
+      {
+        "rule"  => ATTR,
+        "value" => VALUE,
+        "name"  => NAME
+      }
+    ]
+
+    ay_rules = [
+      "SUBSYSTEM==\"net\", ACTION==\"add\", DRIVERS==\"?*\", %s==\"%s\", NAME=\"%s\"" %
```

Comment:
well in this case this template overwrites everything in the target file, so it is expected output.